### PR TITLE
Add Leader Election Lease Options

### LINKED ISSUE #2836: keda operator restarted at the time of start.(error retrieving resource lock keda/operator.keda.sh)
### Discussed in https://github.com/kedacore/keda/discussions/2722

<div type='discussions-op-text'>

<sup>Originally posted by **vkamlesh** March  7, 2022</sup>
Keda operator failed to elect leader after keda-operator pod restart. These restarts are not frequent but it's happening in a few days(6 days) time intervals. 

KEDA Version: 2.6.1
Git Commit: efca71d6bc770408468a9e1a4b3984f7136c0967

Kubernetes version: v1.20.9
Kubernetes Cluster : AKS

```

bash-3.2$ k get po -n keda
NAME                                      READY   STATUS    RESTARTS   AGE
keda-metrics-apiserver-649f4ddbbd-v4pjp   1/1     Running   0          12d
keda-operator-68ddbdcc8f-6h767            1/1     Running   3          12d
bash-3.2$ 


bash-3.2$ kubectl get --raw "/apis/coordination.k8s.io/v1/namespaces/keda/leases/operator.keda.sh"
{"kind":"Lease","apiVersion":"coordination.k8s.io/v1","metadata":{"name":"operator.keda.sh","namespace":"keda","uid":"edb18fd7-b95e-463f-81cf-6a1010073409","resourceVersion":"135421212","creationTimestamp":"2022-02-23T14:33:32Z","managedFields":[{"manager":"keda","operation":"Update","apiVersion":"coordination.k8s.io/v1","time":"2022-02-23T14:33:32Z","fieldsType":"FieldsV1","fieldsV1":{"f:spec":{"f:acquireTime":{},"f:holderIdentity":{},"f:leaseDurationSeconds":{},"f:leaseTransitions":{},"f:renewTime":{}}}}]},"spec":{"holderIdentity":"keda-operator-68ddbdcc8f-6h767_53e840b6-5466-484f-a6f6-16978b7ee12c","leaseDurationSeconds":15,"acquireTime":"2022-03-07T12:07:54.000000Z","renewTime":"2022-03-07T17:36:42.260717Z","leaseTransitions":142}}




bash-3.2$ k logs keda-operator-68ddbdcc8f-6h767 -n keda -f -p


1.6466548397264059e+09	INFO	controller.scaledobject	Reconciling ScaledObject	{"reconciler group": "keda.sh", "reconciler kind": "ScaledObject", "name": "observationsprocessor-func", "namespace": "platform-api"}
E0307 12:07:33.812275       1 leaderelection.go:330] error retrieving resource lock keda/operator.keda.sh: Get "https://10.0.0.1:443/apis/coordination.k8s.io/v1/namespaces/keda/leases/operator.keda.sh": context deadline exceeded
I0307 12:07:33.812329       1 leaderelection.go:283] failed to renew lease keda/operator.keda.sh: timed out waiting for the condition
1.6466548538123553e+09	ERROR	setup	problem running manager	{"error": "leader election lost"}`
```
</div>

### PR DESCRIPTION
**Pull request by wsugarman**
Add new options for configuring the leader election lease, including:
  - `KEDA_OPERATOR_LEADER_ELECTION_LEASE_DURATION`
  - `KEDA_OPERATOR_LEADER_ELECTION_RENEW_DEADLINE`
  - `KEDA_OPERATOR_LEADER_ELECTION_RETRY_PERIOD`
  - `KEDA_METRICS_LEADER_ELECTION_LEASE_DURATION`
  - `KEDA_METRICS_LEADER_ELECTION_RENEW_DEADLINE`
  - `KEDA_METRICS_LEADER_ELECTION_RETRY_PERIOD`

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [X] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts))
  - **Skipped** because these values are new environment variables that may be configured through the existing `env` element
- [X] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Relates to #2836
Relates to https://github.com/kedacore/keda-docs/pull/842

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -84,13 +84,34 @@ func main() {
 		os.Exit(1)
 	}
 
+	leaseDuration, err := kedautil.ResolveOsEnvDuration("KEDA_LEADER_ELECTION_LEASE_DURATION", time.Duration(15)*time.Second)
```
Comment (wsugarman) [inline main.go:87]: Optionally, I could change this to no longer provide a default and rely on `*time.Duration` such that the controller manager instead picks the default. These defaults were chosen based on their docs

Comment (wsugarman) [reply main.go:87]: Instead, we could also move any defaults to the chart

**Comment by wsugarman** [comment]
I am unsure if this should result in new documentation

File: main.go (hunk 1 above)
Comment (zroubalik) [reply main.go:87]: Yeah, I'd actually prefer if we rely on controller manager defaults.

**Comment by zroubalik** [comment]
> I am unsure if this should result in new documentation

Yeah, we should cover this in this section of docs: https://keda.sh/docs/2.8/operate/cluster/

Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,9 +34,11 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 
 ## Unreleased
 
+- **General:** Add settings for configuring leader election ([#2836]https://github.com/kedacore/keda/issues/2836))
```
Comment (zroubalik) [inline CHANGELOG.md:37]: Please move this to Improvements section and sort alphabetically

**Comment by wsugarman** [comment]
Taking another look, perhaps there are too many options and it's better to simply share the same values for both the operator and metrics server?

**Review by zroubalik — APPROVED** [review]
yeah, it's a lot of new values, but I think that it is okay to keep it like this :)

**Comment by wsugarman** [comment]
@zroubalik Thank you for your help!

**Pull request merged**